src/mission_ui.cpp

```diff
@@ -17,239 +17,277 @@
 #include "translations.h"
 #include "ui.h"
 #include "ui_manager.h"
+#include "cata_imgui.h"
+#include "imgui/imgui.h"
 
-void game::list_missions()
+enum class mission_ui_tab_enum : int {
```

Comment:
This would reduce the enum declaration and its operators, but I don't expect major benefits.

I'd like to simplify the huge ImGui::BeginTabBar block into a single `for` loop, but each tab needs a different function to assign `umissions`. I guess we could check the array value immediately after doing the tabs and then enter a switch statement to return the appropriate vector of missions.

I am open to further suggestions/discussions regarding this.

Reply:
True. Perhaps an array of pairs? First the label and then a lambda that calls the getter (or even just a function pointer)? That would be a very traditional solution in Lisp, but C++ does make it slightly more annoying.

Now that I think about it, you could just call all three getters up front. 

```c++
auto labels = [ _("ACTIVE"), _("COMPLETED"), _("FAILED") ];
auto missions = [get_avatar().get_active_missions(),
                 get_avatar().get_completed_missions(),
                 get_avatar().get_failed_missions() ];
```

Reply:
I'd do something like
```cpp
struct tab_data {
    std::string label;
    std::vector<mission *> missions;
}

std::map<mission_ui_tab_enum, tab_data> tabs {
    { mission_ui_tab_enum::ACTIVE, { _( "ACTIVE" ), get_avatar().get_active_missions() },
    ...
};
```
And then access the things by enum. Looping over the enum should also be possible I think? Never tried that, though.
```cpp
for ( mission_ui_tab_enum i = mission_ui_tab_enum::ACTIVE; i < mission_ui_tab_enum::num_tabs; ++i ) {
    // do stuff with i
}
```

Reply:
You technically can do that, but shouldn't; it’ll break as soon as anyone does anything weird with the enum. Two simple arrays accessed by the same integer index is the simplest and most straight–forward thing to do.

Reply:
But why would anyone do anything weird with the enum (and not notice immediately that what they did isn't working)? It'd be far more likely someone does something weird to either or both of the arrays or the integer accessing them. And depending on the weirdness it'd also be more difficult to identify.

Granted this whole use case is pretty simple in general, so either way it's not likely to break in a complicated manner.

Reply:
/me shrugs. I’m just saying that the least complicated code is both the least error–prone and the shortest.